APCAppCore/APCAppCore/UI/TabBarControllers/Profile/APCDemographicUploader.m

```diff
@@ -0,0 +1,125 @@
+static  NSString  *kPatientCurrentAgeKey           = @"patientCurrentAge";
+static  NSString  *kPatientBiologicalSexKey        = @"patientBiologicalSex";
+static  NSString  *kPatientHeightInchesKey         = @"patientHeightInches";
+static  NSString  *kPatientWeightPoundsKey         = @"patientWeightPounds";
+static  NSString  *kPatientWakeUpTimeKey           = @"patientWakeUpTime";
+static  NSString  *kPatientGoSleepTimeKey          = @"patientGoSleepTime";
+
+@implementation APCDemographicUploader
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self != nil) {
+    }
+    return  self;
+}
+
+- (void)uploadNonIdentifiableDemographicData
+{
+    APCUser  *user = ((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.currentUser;
+    
+    NSMutableDictionary  *demographics = [NSMutableDictionary dictionary];
+    
+    [demographics setObject:kNonIdentifiableDemographicsKey forKey:kPatientInformationKey];
+    
+    NSDate  *sleepTime = user.sleepTime;
+    if (sleepTime == nil) {
+        [demographics setObject:[NSNull null] forKey:kPatientGoSleepTimeKey];
+    } else {
+        [demographics setObject:sleepTime forKey:kPatientGoSleepTimeKey];
+    }
+    
+    NSDate  *wakeUpTime = user.wakeUpTime;
+    if (wakeUpTime == nil) {
+        [demographics setObject:[NSNull null] forKey:kPatientWakeUpTimeKey];
+    } else {
+        [demographics setObject:wakeUpTime forKey:kPatientWakeUpTimeKey];
+    }
+    
+    NSDate  *birthDate = user.birthDate;
+    if (birthDate == nil) {
+        [demographics setObject:[NSNull null] forKey:kPatientCurrentAgeKey];
+    } else {
+        NSUInteger  age = [NSDate ageFromDateOfBirth:birthDate];
+        [demographics setObject:[NSNumber numberWithUnsignedInteger:age] forKey:kPatientCurrentAgeKey];
+    }
+    
+    HKBiologicalSex  biologicalSex = user.biologicalSex;
+    NSString  *biologicalSexString = [APCUser stringValueFromSexType:biologicalSex];
+    if (biologicalSexString == nil) {
+        [demographics setObject:[NSNull null] forKey:kPatientBiologicalSexKey];
+    } else {
+        [demographics setObject:biologicalSexString forKey:kPatientBiologicalSexKey];
+    }
+    
+    HKQuantity  *height = user.height;
+    double  heightInInches = [APCUser heightInInches:height];
+    int  klass = fpclassify(heightInInches);
+    if ((klass == FP_INFINITE) || (klass == FP_NAN) || (klass == FP_ZERO)) {
+        [demographics setObject:@(0) forKey:kPatientHeightInchesKey];
+    } else {
+        [demographics setObject:@(heightInInches) forKey:kPatientHeightInchesKey];
+    }
+    
+    HKQuantity  *weight = user.weight;
+    double  weightInPounds = [APCUser weightInPounds:weight];
+    int  klarse = fpclassify(weightInPounds);
+    if ((klarse == FP_INFINITE) || (klass == FP_NAN) || (klass == FP_ZERO)) {
+        [demographics setObject:@(0) forKey:kPatientWeightPoundsKey];
+    } else {
+        [demographics setObject:@(weightInPounds) forKey:kPatientWeightPoundsKey];
```

Comment:
I think modern dictionary access would look nicer throughout this method: 

demographics[key] = <bool check> ? value : [NSNull null];